lua/opencode/cli/server.lua

```diff
@@ -43,53 +56,142 @@ local function find_servers()
-              if port then
-                port = tonumber(port)
-
-                table.insert(
-                  servers,
-                  ---@class Server
-                  ---@field pid number
-                  ---@field port number
-                  ---@field cwd string
-                  {
-                    pid = pid,
-                    port = port,
-                    cwd = cwd,
-                  }
-                )
-              end
+      local lsof_output = exec("lsof -w -iTCP -sTCP:LISTEN -P -n -a -p " .. pid .. " 2>/dev/null || true")
+
+      if lsof_output ~= "" then
+        for line in lsof_output:gmatch("[^\r\n]+") do
+          local parts = vim.split(line, "%s+")
+
+          if parts[1] ~= "COMMAND" then -- Skip header
+            local port = parts[9] and parts[9]:match(":(%d+)$") -- e.g. "127.0.0.1:12345" -> "12345"
+            if port then
+              port = tonumber(port)
+
+              table.insert(processes, {
+                pid = pid,
+                port = port,
+              })
             end
           end
         end
       end
     end
   end
 
+  if #processes == 0 then
+    error("No `opencode` processes found", 0)
+  end
+
+  return processes
+end
+
+---Find running opencode PIDs and ports on Windows using PowerShell
+---@return {pid: number, port: number}[]
+local function retrieveOpencodeProcessesWin()
+  local ps_script = [[
+Get-Process -Name '*opencode*' -ErrorAction SilentlyContinue |
+ForEach-Object {
+  $ports = Get-NetTCPConnection -State Listen -OwningProcess $_.Id -ErrorAction SilentlyContinue
+  if ($ports) {
+    foreach ($port in $ports) {
+      [PSCustomObject]@{pid=$_.Id; port=$port.LocalPort}
+    }
+  }
+} | ConvertTo-Json -Compress
+]]
+
+  -- Execute PowerShell synchronously, but this doesn't hold up the UI since
+  -- this gets called from a function that returns a promise.
+  local ps_result = vim.system({ "powershell", "-NoProfile", "-Command", ps_script }):wait()
+
+  if ps_result.code ~= 0 then
+    error("PowerShell command failed with code: " .. ps_result.code, 0)
+  end
+
+  if not ps_result.stdout or ps_result.stdout == "" then
+    error("No `opencode` processes found", 0)
+  end
+
+  -- The Powershell script should return the response as JSON to ease parsing.
+  local ok, processes = pcall(vim.fn.json_decode, ps_result.stdout)
+  if not ok then
+    error("Failed to parse PowerShell output: " .. tostring(processes), 0)
+  end
+
+  if processes.pid then
+    -- A single process was found, so wrap it in a table.
+    processes = { processes }
+  end
+
+  if #processes == 0 then
+    error("No `opencode` processes found", 0)
+  end
+
+  return processes
+end
+
+-- Find the working directories of running instances of opencode.
+---@param processes {pid: number, port: number}[]
+---@return Server table
+local function populateWorkingDirectories(processes)
+  local servers = {}
+
+  if vim.fn.executable("curl") == 0 then
```

Comment:
Making curl a transversal required tool is debatable: 
- can't a simple and portable LUA http client be used instead? 
- if no, can't we have a fallback on well known HTTP client CLI such as wget? or even lsof usage on *nix?

Reply:
@NickvanDyke - Do you have a viewpoint on this?